### PR TITLE
feat: add support for tailing logs

### LINKED ISSUE #1564: Tailing logs
`osctl logs` & `osctl dmesg` should support `--tail=<lines>` flag.

### PR DESCRIPTION
**Pull request by smira**
Fixes #1564

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: api/machine/machine.proto [hunk 1]
```diff
@@ -28,63 +28,41 @@ service MachineService {
   rpc Shutdown(google.protobuf.Empty) returns (ShutdownResponse);
   rpc Upgrade(UpgradeRequest) returns (UpgradeResponse);
 
-  rpc Start(StartRequest) returns (StartResponse) {
```
Comment (smira) [inline api/machine/machine.proto]: looks like my editor, will revert

File: internal/pkg/tail/tail.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+// Window is the read window size for tail scanning.
+const Window = 4096
+
+// SeekLines seeks the passed io.ReadSeeker so that it's -N lines from the tail.
```
Comment (smira) [inline internal/pkg/tail/tail.go]: couldn't find a better name for it, suggestions?

**Comment by smira** [comment]
TODO: 

[ ] add integration test for tail

File: internal/pkg/tail/tail.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/pkg/tail/tail.go]: I think this sounds fine. It describes what it does.

File: api/machine/machine.proto [hunk 3]
```diff
@@ -258,6 +258,7 @@ message LogsRequest {
   // driver might be default "containerd" or "cri"
   common.ContainerDriver driver = 3;
   bool follow = 4;
+  int32 tailLines = 5;
```
Comment (andrewrynhard) [inline api/machine/machine.proto]: nit: the convention in protobuf is to use snake case.

Comment (smira) [reply api/machine/machine.proto]: thanks, fixed!

**Comment by smira** [comment]
_technically_ this can go into 0.3 as it's backwards compatible change

**Pull request merged**